src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,120 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User firstUser;
+    private User secondUser;
+    private User thirdUser;
+    private User fourthUser;
+
+    @BeforeAll
+    public static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        firstUser = new User("Mike", "mike12345", 22);
+        secondUser = new User("Alice", "password", 19);
+        thirdUser = new User("Melisa", "qwerty", 20);
+        fourthUser = new User("MaX", "MaX24MaX", 24);
+    }
+
+    @Test
+    public void register_passwordLength_isValid() {
+        firstUser.setPassword("dddssssssssad");
+        secondUser.setPassword("asdadado");
+        thirdUser.setPassword("sss123osaa");
+        fourthUser.setPassword("qwrerasfdsdfs");
+        assertEquals(firstUser, registrationService.register(firstUser));
+        assertEquals(secondUser, registrationService.register(secondUser));
+        assertEquals(thirdUser, registrationService.register(thirdUser));
+        assertEquals(fourthUser, registrationService.register(fourthUser));
+    }
+
+    @Test
+    public void register_passwordLength_isNotValid() {
+        firstUser.setPassword("ddd");
+        secondUser.setPassword("0");
+        thirdUser.setPassword(null);
+        fourthUser.setPassword("..");
+        assertThrows(RuntimeException.class, () -> registrationService.register(firstUser));
+        assertThrows(RuntimeException.class, () -> registrationService.register(secondUser));
+        assertThrows(RuntimeException.class, () -> registrationService.register(thirdUser));
+        assertThrows(RuntimeException.class, () -> registrationService.register(fourthUser));
+    }
+
+    @Test
+    public void register_age_isNotValid() {
+        firstUser.setAge(10);
+        secondUser.setAge(null);
+        thirdUser.setAge(17);
+        fourthUser.setAge(-100);
+
+        assertThrows(RuntimeException.class, () -> registrationService.register(firstUser));
+        assertThrows(RuntimeException.class, () -> registrationService.register(secondUser));
+        assertThrows(RuntimeException.class, () -> registrationService.register(thirdUser));
+        assertThrows(RuntimeException.class, () -> registrationService.register(fourthUser));
+    }

```

Comment:
The first part of the test method name should be the name of the method that you test;
The second part should describe the state you pass to this method;
And the last part should describe the expected behavior;
For example: 
register_nullPasword_notOk
register_existingUser_notOk

Let's divide your test methods into several smaller ones with proper names